nixos/modules/services/desktops/gnome/gnome-keyring.nix

```diff
@@ -1,53 +1,52 @@
 # GNOME Keyring daemon.
 
-{ config, pkgs, lib, ... }:
-
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.gnome.gnome-keyring;
+in
 {
 
-  meta = {
-    maintainers = lib.teams.gnome.members;
-  };
-
-  ###### interface
-
-  options = {
-
-    services.gnome.gnome-keyring = {
+  meta.maintainers = lib.teams.gnome.members;
 
-      enable = lib.mkOption {
-        type = lib.types.bool;
-        default = false;
-        description = ''
-          Whether to enable GNOME Keyring daemon, a service designed to
-          take care of the user's security credentials,
-          such as user names and passwords.
-        '';
-      };
+  ## Interface ##
 
+  options.services.gnome.gnome-keyring = {
+    enable = lib.mkEnableOption null // {
+      default = false;
+      description = ''
+        Whether to enable GNOME Keyring daemon, a service designed to
+        take care of the user's security credentials,
+        such as user names and passwords.
+      '';
     };
-
+    package = lib.mkPackageOption pkgs "gnome-keyring" { default = pkgs.gnome.gnome-keyring; };
```

Comment:
Is there as specific reason for adding the package option? Are users expected to change it? If it is just for debugging, maybe overlays would be sufficient?

Reply:
Overlays are pretty complicated for new users. I added the option in case anyone wants to override the gnome-keyring package like use an older version, or patch it themselves.

This package option helps sort that out without resorting to overlays, which can actually rebuild the whole system sometimes (may be not for this package). Overlays are known to cause problems.

As for the `gcrPackage` option, I thought I would use `gcr_4` with this. I assume `gcr` ver 3 is still the default due to compatibility issues but for GNOME 46, shouldn't `gcr_4` be the default?

Reply:
Well, new users should not need to change packages under modules in the first place. If there is a bug, we should fix it in Nixpgks. And the mass rebuilds are not an issue with leaf packages like GNOME Keyring.

`gcr` contains two D-Bus services that were removed in [GCR 4](https://gitlab.gnome.org/GNOME/gcr/-/commit/f316c8919bfcbe2534271e9729084342db3cc9c6). AFAICT, it does not even make sense to install that on GNOME Shell, since it provides its own implementation.

Looks like `gcr` installation was introduced in 2d280840f844cad6c9478bc9107b02cff82b1c34 to provide fallback prompter for GNOME Keyring (well, it uses the prompter through `gcr` library). But that should probably go into its own module, as GNOME Keyring is [not the only software that expects this](https://codesearch.debian.net/search?q=org.gnome.keyring.SystemPrompter&literal=1).

Reply:
Done. Removed `package` and `gcrPackage` option.